src/sql/workbench/contrib/executionPlan/browser/executionPlan.ts

```diff
@@ -362,12 +368,50 @@ export class ExecutionPlan implements ISashLayoutProvider {
 			}
 		}
 
+		if (node.badges) {
+			diagramNode.badges = [];
+			for (let i = 0; i < node.badges.length; i++) {
+				diagramNode.badges.push(this.getBadgeTypeString(node.badges[i].type));
+			}
+		}
+
 		if (node.description) {
 			diagramNode.description = node.description;
 		}
 		return diagramNode;
 	}
 
+	private getBadgeTypeString(badgeType: BadgeType): {
+		type: string,
+		tooltip: string
+	} {
+		/**
+		 * TODO: Need to figure out if tooltip have to be removed. For now, they are empty
+		 */
+		switch (badgeType) {
+			case BadgeType.WARNING:
+				return {
+					type: 'warning',
+					tooltip: ''
+				};
+			case BadgeType.CRITICALWARNING:
+				return {
+					type: 'criticalWarning',
+					tooltip: ''
+				};
+			case BadgeType.PARALLELISM:
+				return {
+					type: 'parallelism',
+					tooltip: ''
+				};
+			default:
+				return {
+					type: 'warning',
```

Comment:
Why would we want to default to a type of warning, if the badge type isn't `WARNING`, `CRITICALWARNING`, or `PARALLELISM`?

Reply:
Yeah, seems like we probably just shouldn't show the badge in that case - it isn't something we should expect to happen.